x-pack/plugins/enterprise_search/server/routes/workplace_search/role_mappings.test.ts

```diff
@@ -128,4 +129,27 @@ describe('role mappings routes', () => {
       });
     });
   });
+
+  describe('POST /api/workplace_search/org/single_user_role_mapping', () => {
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({
+        method: 'post',
+        path: '/api/workplace_search/org/single_user_role_mapping',
+      });
+
+      registerOrgUserRoute({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request handler', () => {
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/ws/org/role_mappings/upsert_single_user_role_mapping',
+      });
+    });
+  });
```

Comment:
I would strongly consider adding validation tests for the body schema you've created in [lines 104-110](https://github.com/elastic/kibana/pull/103173/files#diff-8f9e469453de97d5767ab3fadd02c05fd19f8417e49eb569da8138871c9666c2R104-R110). Just one example of an empty body failing and a full body passing should be fine

Reply:
https://github.com/elastic/kibana/pull/103173/commits/ae4c9bffea13e27ab18ed4a336b61c45fe588021

Reply:
Awesome, looks fantastic!